ahnlich/replication-rfc.md

```diff
@@ -0,0 +1,41 @@
+# Ahnlich replication RFC
+
+The first thing we want to do is to identify the components we need to implement.
+
+They are (NOTE: these names are tentative):
+
+- `LogStore` - This is where the logs from the Raft cluster activities will be stored. Here is an in-memory impl from the openraft guys that they used in their example: <https://github.com/databendlabs/openraft/blob/4f0fd5fa034413d2f367306da4a0016f7603fb7e/examples/mem-log/src/log_store.rs>, and I believe we can easily co-opt this into a write-to-disk-log file service, or any other log storage we settle on.
+
+- `StateMachineStore` - This is where the last known state (snapshot) is stored and read from. They have a neat impl here as well: <https://github.com/databendlabs/openraft/blob/4f0fd5fa034413d2f367306da4a0016f7603fb7e/examples/raft-kv-memstore-grpc/src/store/mod.rs>, and I think the bit we need to figure out are where we want to 'store' the state machine.
```

Comment:
What is a snapshot in this sense with respect to the state? All the application data in the store in some serializable format?